app/services/ad_reminder.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+# A class to implement all logic related to sending of reminder e-mails
+class AdReminder
+  include Rails.application.routes.url_helpers
+  default_url_options[:host] = Rails.application.config.web_host
+  default_url_options[:protocol] = Rails.application.config.web_protocol
+
+  def remind_oldest(batch_size)
+    Ad.with_email.not_deleted.not_reminded.expiring.order(created_at: :asc).limit(batch_size).each do |ad|
+      ad.refresh_token!
+      AdMailer.send_reminder(
+        ad,
+        ad_url(ad),
+        edit_ad_url(ad, t: ad.token),
```

Comment:
Ah I understand now that you needed to set `default_url_options` for these methods, not for ActionMailer.

This is how I had it setup in one of my apps where ActiveStorage is rendering an attachment's partial (ERB file):

```erb
<%= link_to(
      blob.filename,
      rails_blob_url(
        blob,
        Rails.application.config.action_mailer.default_url_options.merge(disposition: "attachment")
      )
    ) %>
```

In this case that would be:

```rb
        edit_ad_url(ad, Rails.application.config.action_mailer.default_url_options.merge(t: ad.token)),
```

...but that also means we'll be relying on ActionMailer `default_url_options`.

Either way, I'm more in favor of shipping this than bikeshedding. :)

Reply:
I went with Vlado's suggestion, please have a look.